PracticeGit/Test.c

```diff
@@ -1,13 +1,151 @@
+// test.c -  Program that Executes test cases and checks the user input for sort & search.
+// ------------------------------------------------------------------------------------------------
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+#define MAXSIZE 10
+#define ELEMENT_NOT_FOUND -1
+#define RED     "\033[1;31m"
+#define GREEN   "\033[1;32m"
+#define RESET   "\033[0m"
+
+/// <summary> Prints the elements of an array. </summary>
+void PrintArray (int array[], int size);
+
+/// <summary> Runs test cases for the QuickSort function. </summary>
+void RunQuickSortTest ();
+
+/// <summary> Gets user input for an array, sorts it, and performs a binary search. </summary>
+void GetUserInput ();
+
+/// <summary>Compare Arrays returns true or false. </summary>
+bool ArraysEqual (int arr1[], int arr2[], int size);
+
+typedef struct {
+   int array[MAXSIZE], size;
+} TestCase;
+
+void PrintArray (int array[], int size) {
+   for (int i = 0; i < size; i++) printf ("%3d", array[i]);
+}
+
+bool ArraysEqual (int arr1[], int arr2[], int size) {
+   for (int i = 0; i < size; i++) {
+      if (arr1[i] != arr2[i]) return false;
+   }
+   return true;    // Arrays are equal if all elements match
+}
+
+void RunQuickSortTest () {
+   TestCase testCases[] = {
+       {{20, 49, 11, 3, 89, 75, 11, 12, 1}, 9},
+       {{2, 2, 2, 2, 2, 2, 2}, 7},
+       {{9}, 1},
+       {{3, -1, -7, 5, 2, 0, -7}, 7},
+       {{1, 2, 3, 4, 5, 6}, 6},
+       {{0, -1, -2, -3, -4, -5}, 6},
+       {{36, 69}, 2},
+       {{22, 35, 32, 39, 46, 54, 55}, 7},
+       {{12, 13, 14, 15, 16, 11}, 6}
+   };
+   int expectedResults[][MAXSIZE] = {
+       {1, 3, 11, 11, 12, 20, 49, 75, 89},
+       {2, 2, 2, 2, 2, 2, 2},
+       {9},
+       {-7, -7, -1, 0, 2, 3, 5},
+       {1, 2, 3, 4, 5, 6},
+       {-5, -4, -3, -2, -1, 0},
+       {36, 69},
+       {22, 32, 35, 39, 46, 54, 55},
+       {11, 12, 13, 14, 15, 16}
+   };
+   int elementSearch[] = { 11, 1, 0, -1, 2, 6, 36, 53, 11 };
+   int testCount = sizeof (testCases) / sizeof (testCases[0]);
+   printf ("+--------------------------------------------------+---------------+------------------+--------+\n"
+      "|                Arrays                            | Key to Search | Index of element | Result |\n"
+      "+--------------------------------------------------+---------------+------------------+--------+\n");
+   for (int i = 0; i < testCount; i++) {
+      int size = testCases[i].size, * arrayToSort = testCases[i].array;
+      printf ("| Unsorted Array   | ");
+      PrintArray (arrayToSort, size);
+      printf ("%*s|", (MAXSIZE - size) * 3, " ");
+      int searchValue = elementSearch[i];   // Element to search for in the current test case
+      QuickSort (arrayToSort, 0, size - 1);
+      int foundIndex = BinarySearch (arrayToSort, size, searchValue);
+      if (foundIndex != ELEMENT_NOT_FOUND) printf ("    %-11d| Found at index %-d |", searchValue, foundIndex);
```

Comment:
Meaningful way to write this is :

```suggestion
      if (foundIndex == ELEMENT_NOT_FOUND) printf ("    %-11d|    Not Found     |", searchValue);
      else printf (" %-11d| Found at index %-d |", searchValue, foundIndex);
```